fenix/app/src/main/java/org/mozilla/fenix/translations/preferences/downloadlanguages/DeleteLanguageFileDialogPreference.kt

```diff
@@ -0,0 +1,126 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.translations.preferences.downloadlanguages
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.AlertDialog
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import mozilla.components.feature.downloads.toMegabyteOrKilobyteString
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.annotation.LightDarkPreview
+import org.mozilla.fenix.compose.button.TextButton
+import org.mozilla.fenix.theme.FirefoxTheme
+import java.util.Locale
+
+/**
+ * Firefox Preference: Download Languages Delete Dialog compose view.
+ *
+ * @param language Language name that should be displayed in the dialogue title.
+ * @param isAllLanguagesItemType Whether the download language file item is of type all languages.
+ * @param fileSize Language file size in bytes that should be displayed in the dialogue title.
+ * @param onConfirmDelete Invoked when the user clicks on the "Delete" dialog button.
+ * @param onCancel Invoked when the user clicks on the "Cancel" dialog button.
+ */
+@Composable
+fun DeleteLanguageFileDialogPreference(
+    language: String,
+    isAllLanguagesItemType: Boolean,
+    fileSize: Long,
+    onConfirmDelete: () -> Unit,
+    onCancel: () -> Unit,
+) {
+    AlertDialog(
+        onDismissRequest = {},
+        modifier = Modifier.background(
+            color = FirefoxTheme.colors.layer2,
+            shape = RoundedCornerShape(8.dp),
+        ),
+        title = {
+            val title: String = if (isAllLanguagesItemType) {
+                stringResource(
+                    id = R.string.delete_language_all_languages_file_dialog_title,
+                    fileSize.toMegabyteOrKilobyteString(),
+                )
+            } else {
+                stringResource(
+                    R.string.delete_language_file_dialog_title,
+                    language,
+                    fileSize.toMegabyteOrKilobyteString(),
```

Comment:
nit: Add named parameters please. Also double check this elsewhere

Reply:
Done